content/hello-world-program.md

```diff
@@ -20,30 +20,91 @@ Solana's ability to run arbitrary executable code is part of what makes it so po
+Before we dive into the building our "Hello world!" program, let’s first go over some Rust basics. If you want to dig deeper into Rust, have a look at the [Rust language book](https://doc.rust-lang.org/book/ch00-00-introduction.html).
+
+### Module System
+
+Rust organizes code using what is collectively referred to as the “module system”. 
+
+This includes:
+
+- **Packages** - A Cargo feature that lets you build, test, and share crates
+- **Crates** - A tree of modules that produces a library or executable
+- **Modules** and **use** - Let you control the organization, scope, and privacy of paths
+- **Paths** - A way of naming an item, such as a struct, function, or module
 
-To write Solana programs with Rust, we use the [solana_program](https://docs.rs/solana-program/latest/solana_program/index.html) library crate. Crates in Rust define functionality that can be shared with multiple projects. The `solana_program` crate acts as a standard library for Solana programs. This standard library contains the modules and macros that we'll use to develop our Solana programs. You can read more about Rust crates [here](https://doc.rust-lang.org/book/ch07-01-packages-and-crates.html).
+For this lesson, we’ll focus on using crates, modules, and paths.
 
 ### Pathways and scope
 
-To tell Rust where to find an item in a module tree, we use a path just like when we're navigating a filesystem. If we want to call a particular function within a module, then we need to know the pathway to it.
+Crates in Rust contain modules that define functionality which can be shared with multiple projects.  If we want to access an item within a module, then we need to know the path (like when we're navigating a filesystem).
 
-To "use" a module available within the `solana_program` crate we'll need to know:
+To use a module available within a crate, we'll need to:
 
-1. How to invoke the module by its path
-2. How bring it into the scope of our program.
+1. Bring the module into the scope of our program
+2. Specify the item in the module tree using its path
 
-Paths are brought into scope with the [use](https://doc.rust-lang.org/stable/book/ch07-04-bringing-paths-into-scope-with-the-use-keyword.html) keyword. In the example below, we bring into scope the [AccountInfo](https://docs.rs/solana-program/latest/solana_program/account_info/struct.AccountInfo.html) struct from the `account_info` module within the `solana_program` crate.
+Paths are brought into scope with the [use](https://doc.rust-lang.org/stable/book/ch07-04-bringing-paths-into-scope-with-the-use-keyword.html) keyword. In the example below, we bring into scope the `AccountInfo` struct from the `account_info` module within the `solana_program` crate. Bringing an item into scope this way allows us to use the item multiple times within our program without specifying the full path each time.
```

Comment:
Could we add something about the `::` double colon? That's fairly unique to Rust so it'll be new for some people. Probably worth explaining the difference between that and just a dot. @ZYJLiu

Reply:
updated with explanation for "::"
but wasn't sure where to include explanation and example for "." 

the hello world program doesn't use "." to call any methods
where would you like the explanation to go? (in its own section or mention as part of path/scope)
 
 https://doc.rust-lang.org/reference/expressions/method-call-expr.html#method-call-expressions

Reply:
It uses `.` for calling properties though, which in a lot of languages is the same for calling properties and functions...but I'm fine leaving it out.